pkg/aws/sts.go

```diff
@@ -262,10 +263,23 @@ func (c *awsClient) ListOidcProviders(targetClusterId string) ([]OidcProviderOut
 			if skip {
 				continue
 			}
-			providers = append(providers, OidcProviderOutput{
-				Arn:       *provider.Arn,
-				ClusterId: clusterId,
-			})
+			if idFilter != "" {
+				resourceId, err := GetResourceIdFromOidcProviderARN(*provider.Arn)
+				if err != nil {
+					return nil, fmt.Errorf("unable to get resource ID from OIDC Provider's ARN. Error: '%v'", err)
+				}
+				if strings.Contains(resourceId, idFilter) {
```

Comment:
You could move this prior to the skip and consider this is a skip true when it doesn't contain instead. So that you can keep appending only once at the end of the flow